test/fail_compilation/test17892.d

```diff
@@ -0,0 +1,28 @@
+/* REQUIRED_ARGS: -dip1000
+TEST_OUTPUT:
+---
+fail_compilation/test17892.d(25): Error: returning `s.pointer1()` escapes a reference to local variable `s`
+fail_compilation/test17892.d(27): Error: returning `s.pointer2()` escapes a reference to local variable `s`
+---
+*/
+
+
+// https://issues.dlang.org/show_bug.cgi?id=17892
+
+struct S
```

Comment:
Why isne the struct templated?

Reply:
because the problem is with inference, and `auto` will do that just as well and is simpler.